src/connector.js

```diff
@@ -0,0 +1,612 @@
+					.replace( '%[result.printableUri]', result.printableUri)
+					.replace( '%[short-date-en]', resultDate.getFullYear() + "-" + resultDate.getMonth() + 1 + "-" + resultDate.getDate() )
+					.replace( '%[short-date-fr]', resultDate.getFullYear() + "-" + resultDate.getMonth() + 1 + "-" + resultDate.getDate() )
+					.replace( '%[long-date-en]', monthsEn[ resultDate.getMonth() ] + " " + resultDate.getDate() + ", " + resultDate.getFullYear() )
+					.replace( '%[long-date-fr]', resultDate.getDate() + " " + monthsFr[ resultDate.getMonth() ] + " " + resultDate.getFullYear() )
+					.replace( '%[highlightedExcerpt]', highlightedExcerpt );
+				
+				const interactiveResult = buildInteractiveResult( headlessEngine, {
+					options: { result },
+				} );
+
+				let resultLink = sectionNode.querySelector( ".result-link" );
+				resultLink.onclick = ( e ) => { interactiveResult.select(); };
+				resultLink.oncontextmenu = ( e ) => { interactiveResult.select(); };
+				resultLink.onmousedown = ( e ) => { interactiveResult.select(); };
+				resultLink.onmouseup = ( e ) => { interactiveResult.select(); };
+				resultLink.ontouchstart = ( e ) => { interactiveResult.beginDelayedSelect(); };
+				resultLink.ontouchend = ( e ) => { interactiveResult.cancelPendingSelect(); };
+
+				resultListElement.appendChild( sectionNode );
+			});
+		}
+		else if ( resultListState.firstSearchExecuted && !resultListState.hasResults && !resultListState.hasError ) {
+			resultListElement.innerHTML = noResultTemplateHTML;
+		}
+	}
+}
+
+// Update heading that has number of results displayed
+function updateQuerySummaryState( newState ) {
+	querySummaryState = newState;
+	
+	if ( resultListState.firstSearchExecuted && !querySummaryState.isLoading && !querySummaryState.hasError ) {
+		querySummaryElement.textContent = "";
+		if ( querySummaryState.total > 0 ) {
+			let numberOfResults = querySummaryState.total.toString();
+
+			if ( numberOfResults.length > 3 ) {
+                numberOfResults = numberOfResults.substring( 0, numberOfResults.length - 3 ) + ',' + numberOfResults.substring( numberOfResults.length - 3 );
+            }
+			if ( numberOfResults.length > 7 ) {
+                numberOfResults = numberOfResults.substring( 0, numberOfResults.length - 7 ) + ',' + numberOfResults.substring( numberOfResults.length - 7 );
+            }
+
+            querySummaryElement.innerHTML = ( ( querySummaryState.query != "" && !params.isAdvancedSearch ) ? querySummaryTemplateHTML : noQuerySummaryTemplateHTML )
+				.replace( '%[numberOfResults]', numberOfResults )
+				.replace( '%[query]', querySummaryState.query );
+		}
+	}
+	else if ( querySummaryState.hasError ) {
+		querySummaryElement.textContent = "";
+		querySummaryElement.innerHTML = resultErrorTemplateHTML;
+	}
+}
+
+// Update pagination
+function updatePagerState( newState ) {
+	pagerState = newState;
+	pagerElement.textContent = "";
+	let innerHTML = "";
```

Comment:
This variable is not used, please remove
```suggestion
```

Reply:
This is a note for @bblaisATcoveo for his subsequent PR